modules/heat_conduction/src/actions/MortarGapHeatTransferAction.C

```diff
@@ -256,22 +259,20 @@ MortarGapHeatTransferAction::addConstraints()
 
   if (!_user_provided_gap_flux_models)
   {
-
     std::vector<UserObjectName> uoname_strings(0);
-    unsigned int conduction_index = 0;
-    unsigned int radiation_index = 0;
 
     for (const auto & uo_name : _gap_flux_models)
     {
       if (uo_name == MortarGapHeatTransfer::UserObjectToBuild::CONDUCTION)
-        uoname_strings.push_back("gap_flux_model_conduction_object_" +
-                                 MooseUtils::shortName(name()) + "_" +
-                                 Moose::stringify(conduction_index++));
+        uoname_strings.push_back(
+            "gap_flux_model_conduction_object_" + MooseUtils::shortName(name()) + "_" +
```

Comment:
Why is the short name not enough for disambiguation? Shouldn't there be one mortar contact syntax block per primary-secondary boundary pair?

Reply:
If you think someone someday might want multiple radiation or conduction uos, then it makes sense to me to keep things as you've done them here. I just wanted to understand a little bit more for my own edification

Reply:
I was thinking; does it make more sense to have this be a class member as opposed to a file static? Again the `shortName()` grants uniqueness between mortar interfaces. The only time you have ambiguity is if you need to create multiple conduction/radiation uos on a per-mortar-interface basis, right?

Reply:
> The only time you have ambiguity is if you need to create multiple conduction/radiation uos on a per-mortar-interface basis, right?

Yes, that's what I was trying to say above. That the design should allow for having multiple uos per unit of mortar interface.
I'll create another PR using the short name.

Reply:
Counter are often static members in some module actions. I don't remember having used members to account for multiple action instantiations.

Reply:
https://github.com/idaholab/moose/pull/23876